### PR TITLE
Fix: dobbo-go issue #1454

### PR DESCRIPTION
**Pull request by ChangedenCZD**
**What this PR does**:
Optimizing the registration go struct name (combine go PkgPath and go Name).

**Which issue(s) this PR fixes**:

Fixes https://github.com/apache/dubbo-go/issues/1454

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ChangedenCZD** [comment]
@wongoo 如果不考虑JavaClassName重复注册的话，dubbo-go无须变动

**Comment by wongoo** [comment]
> @wongoo 如果不考虑JavaClassName重复注册的话，dubbo-go无须变动

暂不考虑，重复JavaClassName的问题应该在象模型设计上就做好区分。

**Comment by ChangedenCZD** [comment]
涉及的面有点多，暂时先关闭

File: pojo.go [hunk 1]
```diff
@@ -240,6 +241,24 @@ func unRegisterPOJO(o POJO) int {
 	return -1
 }
 
+func getGoName(o interface{}) string {
+	goType := reflect.TypeOf(o)
+	if reflect.Ptr == goType.Kind() {
```
Comment (wongoo) [inline pojo.go:246]: change `if` to `for`

File: pojo.go [hunk 2]
```diff
@@ -240,6 +241,24 @@ func unRegisterPOJO(o POJO) int {
...
+		goType = goType.Elem()
+	}
+	return combineGoName(goType)
+}
+
+func combineGoName(t reflect.Type) string {
+	pkgPath := t.PkgPath()
+	goName := t.String()
+	if pkgPath == "" ||
+		regexp.MustCompile(`^(github\.com/apache/dubbo-go-hessian2|time)`).Match([]byte(pkgPath)) {
```
Comment (wongoo) [inline pojo.go:256]: init regexp as a global variabble

**Review by wongoo — APPROVED** [review]

**Comment by AlexStocks** [comment]
Pls add unit tests for this pr. maybe some uts for the two new funcs.

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**